.github/workflows/build-pull-request.yml

```diff
@@ -26,6 +26,11 @@ jobs:
     with:
       cache-read-only: true
 
+  test-native:
+    uses: ./.github/workflows/reusable-native-tests.yml
+    with:
+      only-test-native-label: ${{ github.event.label.name == 'test native' }}
```

Comment:
I don't know enough about github actions to say whether this works or not, but just want to point out that existing tests do it a bit differently for example `skip-openj9-tests: ${{ !contains(github.event.pull_request.labels.*.name, 'test openj9') }}` Also `only-test-native-label` should be used inside the reusable flow like here https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/c82d74dc68ac88c45b434d6d5ce09eeb20c5e900/.github/workflows/build-common.yml#L12 and you also need to skip the job if the condition is not met, something like https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/c82d74dc68ac88c45b434d6d5ce09eeb20c5e900/.github/workflows/reusable-test-indy.yml#L26

Reply:
Also you initially also had these tests run when the spring boot label was set, idk whether you removed this deliberately or forgot to add it back.